buildpacks/jvm/src/errors.rs

```diff
@@ -138,5 +112,71 @@ pub(crate) fn on_error_jvm_buildpack(error: OpenJdkBuildpackError) {
             "Could not download OpenJDK distribution.",
             error,
         ),
+        OpenJdkBuildpackError::UnsupportedOpenJdkVersion(artifact_requirement) => {
+            match artifact_requirement.version {
+                HerokuOpenJdkVersionRequirement::Major(major_version) => log_error(
+                    "Unsupported OpenJDK version",
+                    formatdoc! {"
+                        The OpenJDK major version {major_version} you specified in your system.properties file is not supported.
+                        Please specify a supported major version in your system.properties file.
+                    ", major_version = major_version },
+                ),
+                HerokuOpenJdkVersionRequirement::Specific(version) => log_error(
+                    "Unsupported OpenJDK version",
+                    formatdoc! {"
+                        The OpenJDK version {version} you specified in your system.properties file is not supported.
+                        Please specify a supported version in your system.properties file.
+
+                        We recommend specifying only the major version in system.properties.
+                        This will cause the buildpack to always install the latest version of the chosen major version.
+                    ", version = version },
+                )
+            }
+        }
+        OpenJdkBuildpackError::UnsupportedOsError(error) => log_error(
+            "Unsupported Operating System",
+            formatdoc! {"
+                The target operating system for this build is not supported. This error should
+                never occur to users of this buildpack.
+
+                If you see this error, please file an issue:
+                https://github.com/heroku/buildpacks-jvm/issues/new
+
+                Details: {error}
+            ", error = error },
+        ),
+        OpenJdkBuildpackError::UnsupportedArchError(error) => log_error(
+            "Unsupported System Architecture",
+            formatdoc! {"
+                The target system architecture for this build is not supported. This error should
+                never occur to users of this buildpack.
+
+                If you see this error, please file an issue:
+                https://github.com/heroku/buildpacks-jvm/issues/new
+
+                Details: {error}
+            ", error = error },
+        ),
+        OpenJdkBuildpackError::ParseInventoryError(error) => log_error(
+            "Invalid Inventory File",
+            formatdoc! {"
+                The inventory of OpenJDK distributions could not be parsed. This error should
+                never occur to users of this buildpack and is almost always a buildpack bug.
```

Comment:
(optional) For this type of "this could never happen in practice, since any single CI test would have failed" case, would `.expect()` be a better fit, to save having yet more supposedly user-facing error messages (that need eg CX review at some point) that will never be seen?